src/core/main/ts/api/dom/DOMUtils.ts

```diff
@@ -285,12 +285,13 @@ export function DOMUtils(doc: Document, settings: Partial<DOMUtilsSettings> = {}
   };
 
   const get = (elm: Node | string): HTMLElement => {
-    if (elm && doc && typeof elm === 'string') {
-      const node = doc.getElementById(elm);
+    const root = settings.root_element || doc;
+    if (elm && root && typeof elm === 'string') {
+      const node = root.querySelector(`#${elm}`) as HTMLElement;
```

Comment:
Hi @mdgbayly,

I'm simplifying this function in #5758, which will remove the `getElementsByName` call. The `getElementById` call is still there.

As far as I can tell, `getElementById` is supposed to be valid on a ShadowRoot. Are there browsers that don't support this? I'm having a bit of trouble finding reliable info on this.

Reply:
I did a quick test on my Mac, on Firefox 77, Safari 13.1 and Chrome 83 with the following code in the console:

`document.createElement('div').attachShadow({mode: 'open'}).getElementById`

All 3 browsers seem to have this function. Maybe older versions were missing it?

Reply:
A better test snippet:

```
var x = document.createElement('div').attachShadow({mode: 'open'}); var y = document.createElement('div'); y.id = 'myid'; x.appendChild(y); x.getElementById('myid');
```

Reply:
Yeh, if that is supported widely it is most likely a better way to do it. That code snippet in my PR may well have come from an earlier PR from another contributor that I pulled a lot of my ideas from, and I may not have thought to check if shadow roots support getElementById.

Looking at the spec it seems to come via ShadowRoot ->DocumentFragment->NonElementParentNode

https://dom.spec.whatwg.org/#interface-nonelementparentnode

Docs for DocumentFragment.getElementById seem to be missing from MDN
https://developer.mozilla.org/en-US/docs/Web/API/DocumentFragment

Reply:
For now, I think we'll close this PR. If we hit problems with browser support, we'll revisit it.